modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/components/PageRenderer/Actions.es.js

```diff
@@ -0,0 +1,157 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {ClayButtonWithIcon} from '@clayui/button';
+import {ClayDropDownWithItems} from '@clayui/drop-down';
+import React, {
+	createContext,
+	forwardRef,
+	useContext,
+	useEffect,
+	useState,
+} from 'react';
+
+import {EVENT_TYPES} from '../../actions/eventTypes.es';
+import {useForm} from '../../hooks/useForm.es';
+
+const ActionsContext = createContext({
+	activeField: '',
+	hoveredField: '',
+	setActiveField: () => {},
+	setHoveredField: () => {},
+});
+
+/**
+ * ActionsContext is responsible for store which field is being hovered or active
+ */
+export const ActionsProvider = ({children}) => {
+	const [hoveredField, setHoveredField] = useState('');
+	const [activeField, setActiveField] = useState('');
+	const dispatch = useForm();
+
+	useEffect(() => {
+		if (hoveredField) {
+			return dispatch({
+				payload: {fieldName: hoveredField},
+				type: EVENT_TYPES.FIELD_HOVERED,
+			});
+		}
+	}, [hoveredField, dispatch]);
```

Comment:
@diegonvs it would be nice to add a comment about it here because we want to remove it later.